scripting/practicemode/grenade_mysql.sp

```diff
@@ -0,0 +1,264 @@
+  char[] new_map = new char[buffer_len];
+
+  SQL_EscapeString(g_Database, tableName, new_map, buffer_len);
+
+  Format(query, sizeof(query), "CREATE TABLE IF NOT EXISTS `%s` ("
+    ... "`id` int(4) NOT NULL AUTO_INCREMENT,"
+    ... "`steamId` varchar(128) DEFAULT NULL,"
+    ... "`steamName` varchar(128) DEFAULT NULL,"
+    ... "`grenadeId` int(4) NOT NULL,"
+    ... "`name` varchar(128) DEFAULT NULL,"
+    ... "`categories` varchar(128) DEFAULT NULL,"
+    ... "`origin` varchar(128) DEFAULT NULL,"
+    ... "`angles` varchar(128) DEFAULT NULL,"
+    ... "`grenadeType` varchar(128) DEFAULT NULL,"
+    ... "`grenadeOrigin` varchar(128) DEFAULT NULL,"
+    ... "`grenadeVelocity` varchar(128) DEFAULT NULL,"
+    ... "`description` varchar(256) DEFAULT NULL,"
+    ... "`timestamp` datetime DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,"
+    ... "PRIMARY KEY (`grenadeId`),"
+    ... "KEY `id` (`id`))", tableName);
+
+  if (!SQL_FastQuery(g_Database, query)) {
+    char error[255];
+    SQL_GetError(g_Database, error, sizeof(error));
+    PrintToServer("Failed to create table for %s (error: %s)", tableName, error);
+  }
+
+  //Begin massive looping
+  g_GrenadeLocationsKv.Rewind();
+  g_GrenadeLocationsKv.GotoFirstSubKey(false);
+
+  int userCount = 0;
+  do 
+  {
+    userCount += 1;
+    g_GrenadeLocationsKv.GetSectionName(steamId, sizeof(steamId));
+    g_GrenadeLocationsKv.GetString("name", steamName, sizeof(steamName));
+    if (g_GrenadeLocationsKv.GotoFirstSubKey(false)) {
+      g_GrenadeLocationsKv.GotoNextKey(false);
+      int grenadeCount = 0;
+      do
+      {
+        grenadeCount += 1;
+        g_GrenadeLocationsKv.GetSectionName(grenadeId, sizeof(grenadeId));
+        g_GrenadeLocationsKv.GetString("name", name, sizeof(name));
+        g_GrenadeLocationsKv.GetString("categories", categories, sizeof(categories));
+        g_GrenadeLocationsKv.GetString("origin", origin, sizeof(origin));
+        g_GrenadeLocationsKv.GetString("angles", angles, sizeof(angles));
+        g_GrenadeLocationsKv.GetString("grenadeType", grenadeType, sizeof(grenadeType));
+        g_GrenadeLocationsKv.GetString("grenadeOrigin", grenadeOrigin, sizeof(grenadeOrigin));
+        g_GrenadeLocationsKv.GetString("grenadeVelocity", grenadeVelocity, sizeof(grenadeVelocity));
+        g_GrenadeLocationsKv.GetString("description", description, sizeof(description));
+        //$$$ Big money function right here $$$
+        UpsertGrenade(tableName, steamId, steamName, StringToInt(grenadeId), name, categories, origin, angles, grenadeType, grenadeOrigin, grenadeVelocity, description);
+      } while (g_GrenadeLocationsKv.GotoNextKey(false));
+      PrintToServer("%i grenades by %s (%s) were exported.", grenadeCount, steamName, steamId);
+    }
+
+    g_GrenadeLocationsKv.GoBack();
+  } while (g_GrenadeLocationsKv.GotoNextKey(false));
+  PrintToServer("%i users were iterated.", userCount);
+
+  CloseDatabaseConnection();
+
+}
+
+DBStatement exportQuery = null;
```

Comment:
Can you make this a function local? I think declaring it static will accomplish your goal of only initializing it once.